R/params.R

```diff
@@ -0,0 +1,329 @@
+
+knit_params_get <- function(input_lines, params) {
+  # check for recent enough knitr
```

Comment:
Do you need to check for a more recent knitr here? (to avoid the issue of `class` not being available in the params structure).